src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -492,4 +516,24 @@ public static int getDefaultConfigVersion() {
     public AtomicBoolean getInstallDefaultConfig() {
         return installDefaultConfig;
     }
+
+    private class AccessControllerWrappedThread extends Thread {
```

Comment:
I don't think this is needed, I've created a PR [1] that simplifies how the initialization task is triggered so it only happens once, take a look and see if it addresses some of the problems you were tackling with this class.

- [1] https://github.com/cwperks/security/pull/17/files?diff=unified&w=1